app/src/test/java/org/akvo/flow/service/bootstrap/SurveyMapperTest.kt

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2020 Stichting Akvo (Akvo Foundation)
+ *
+ * This file is part of Akvo Flow.
+ *
+ * Akvo Flow is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.akvo.flow.service.bootstrap
+
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.spyk
+import io.mockk.verify
+import org.akvo.flow.domain.Survey
+import org.akvo.flow.domain.SurveyGroup
+import org.akvo.flow.domain.SurveyMetadata
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class SurveyMapperTest {
+
+    @MockK
+    lateinit var surveyMetadata: SurveyMetadata
+
+    @MockK
+    lateinit var surveyGroup: SurveyGroup
+
+    @MockK
+    lateinit var survey: Survey
+
+    private lateinit var surveyMapper: SurveyMapper
+
+    @Before
+    fun setUp() {
+        MockKAnnotations.init(this)
+        surveyMapper = spyk(SurveyMapper())
+        every { (surveyMetadata.id) }.returns("1234")
+        every { (surveyMetadata.version) }.returns(1.0)
+        every { (surveyMetadata.name) }.returns("form1")
+        every { (surveyMetadata.surveyGroup) }.returns(surveyGroup)
+        every { (survey.setLocation(any())) }.returns(Unit)
+        every { (survey.setFileName(any())) }.returns(Unit)
+        every { (survey.name) }.returns("name")
+        every { (survey.setName(any())) }.returns(Unit)
+        every { (survey.setSurveyGroup(any())) }.returns(Unit)
+        every { (survey.setVersion(any())) }.returns(Unit)
+    }
+
+    @Test
+    fun getSurveyIdFromFilePathShouldReturnEmptyIfFolderMissing() {
+        val surveyId = surveyMapper.getSurveyIdFromFilePath("file.xml")
+        assertEquals("", surveyId)
+    }
+
+    //TODO: separate into several methods
+    @Test
+    fun getSurveyIdFromFilePathShouldReturnCorrectIdFromFolder() {
+        var surveyId = surveyMapper.getSurveyIdFromFilePath("123/file.xml")
+        assertEquals("123", surveyId)
```

Comment:
I expected to get 1234 here in relation to https://github.com/akvo/akvo-flow-mobile/pull/1716/files#diff-712dd749f6a46fcacbf07e48bd9d8f403c91161441802a27bca2277070a7b8bfR54

Reply:
I don't understand how one can get 1234, it parses the name and extracts the survey id which is part of the path

Reply:
no here we don't use the mocked one, we use the file path provided, ie "123/form.xml" means the surveyId is 123